application-glossary-machine-translation/application-glossary-machine-translation-api/src/main/java/org/xwiki/contrib/glossary/machineTranslation/internal/DefaultTranslationGlossaryManager.java

```diff
@@ -0,0 +1,142 @@
+                    + "where glossaryObj.name = sourceDoc.fullName "
+                    + "and glossaryObj.name = targetDoc.fullName "
+                    + "and glossaryObj.className = 'Glossary.Code.GlossaryClass' "
+                    + "and sourceDoc.space = 'Glossary' "
+                    + "and targetDoc.space = 'Glossary' "
+                    + "and ((sourceDoc.defaultLanguage = :sourceLanguage and sourceDoc.language = '') "
+                    + "     or sourceDoc.language = :sourceLanguage) "
+                    + "and ((targetDoc.defaultLanguage = :targetLanguage and targetDoc.language = '') "
+                    + "     or targetDoc.language = :targetLanguage)",
+                Query.HQL).bindValue("sourceLanguage", sourceLanguage.toString())
+            .bindValue("targetLanguage", targetLanguage.toString()).execute();
+
+        Map<String, String> glossaryEntries = new HashMap<>();
+        for (Object[] entry : glossaryRawEntries) {
+            String trimmedKey = ((String) entry[0]).trim();
+            String trimmedValue = ((String) entry[1]).trim();
+            glossaryEntries.put(trimmedKey, trimmedValue);
+        }
+        return glossaryEntries;
+    }
+
+    @Override
+    public void synchronizeGlossaries()
+    {
+        Translator translator = translatorManager.getTranslator();
+        XWikiContext context = xwikiContextProvider.get();
+
+        try {
+            List<LocalePairs> translatorSupportedGlossaries = translator.getGlossaryLocalePairs();
+            logger.debug("Fetched the list of supported glossary language combinations : [{}]",
+                translatorSupportedGlossaries);
+
+            // Get the list of languages that are supported in the XWiki preferences
+            DocumentReference reference = new DocumentReference(context.getWikiId(), "XWiki", "XWikiPreferences");
+            XWikiDocument preferencesDoc = context.getWiki().getDocument(reference, context);
+            BaseObject preferencesObject =
+                preferencesDoc.getXObject(new ObjectReference("XWiki.XWikiPreferences", reference));
+            String[] missingLanguagesRaw = preferencesObject.getStringValue("languages").split(",");
+
+            List<GlossaryUpdateEntry> updateEntries = new ArrayList<>();
+
+            for (String sourceLangStr : missingLanguagesRaw) {
+                Locale sourceLanguage = new Locale(sourceLangStr);
+                for (String targetLangStr : missingLanguagesRaw) {
+                    Locale targetLanguage = new Locale(targetLangStr);
+                    String deeplSrcLang = translator.normalizeLocale(sourceLanguage);
+                    String deeplDstLang = translator.normalizeLocale(targetLanguage);
+
+                    // Check that src lang dans dstLang are not identical in case of the language is 'fr_FR' to
+                    // 'fr_CH' which is same for deepL
+                    boolean findMatchingGlossary = translatorSupportedGlossaries.stream()
```

Comment:
What about naming the vaiable "foundMatchingLocalePairs" instead to make it more self explanatory?

Reply:
Fixed by 67756a1